src/server/campaignd/server.cpp

```diff
@@ -1893,15 +2070,22 @@ void server::handle_delete(const server::request& req)
 		return;
 	}
 
-	if(!addon["forum_auth"].to_bool()) {
-		if(!authenticate(*addon, pass)) {
+	if(erase["admin"].to_bool()) {
+		if(!authenticate_admin(erase["uploader"].str(), pass.str())) {
 			send_error("The passphrase is incorrect.", req.sock);
 			return;
 		}
 	} else {
-		if(!authenticate_forum(erase, pass, true)) {
-			send_error("The passphrase is incorrect.", req.sock);
-			return;
+		if(!addon["forum_auth"].to_bool()) {
```

Comment:
I'd avoid negation if possible and switch with the else case.

Reply:
Updated.